java-11/Dockerfile

```diff
@@ -49,30 +49,45 @@ FROM gcr.io/distroless/java-base-debian12:debug AS ssl
 FROM amd64/debian:bookworm-slim As builder
 RUN apt update && apt install -y libjemalloc-dev
 
-# Debian-12 image
-FROM gcr.io/distroless/cc-debian12:debug
-
-MAINTAINER Hypertrace "https://www.hypertrace.org/"
+# stage for amd64
+FROM gcr.io/distroless/cc-debian12:debug AS base-amd64
 
 SHELL ["/busybox/sh", "-c"]
-
 RUN ln -s /busybox/sh /bin/sh
 
-COPY --from=ssl /etc/ssl/certs/java /etc/ssl/certs/java
-
 COPY --from=jre /lib/x86_64-linux-gnu/libz.so.1.2.11 /lib/x86_64-linux-gnu/libz.so.1.2.11
 RUN ln -s /lib/x86_64-linux-gnu/libz.so.1.2.11 /lib/x86_64-linux-gnu/libz.so.1
 
-COPY --from=jre /jre /usr/lib/jvm/zulu-11-amd64-slim
-RUN ln -s /usr/lib/jvm/zulu-11-amd64-slim/bin/java /usr/bin/java
-RUN ln -s /usr/lib/jvm/zulu-11-amd64-slim/bin/jfr /usr/bin/jfr
-
-# change rocksdb default memory allocator
-# https://docs.confluent.io/platform/current/streams/developer-guide/memory-mgmt.html#rocksdb
+# change default memory allocator to jemalloc
 COPY --from=builder /usr/lib/x86_64-linux-gnu/libjemalloc* /usr/lib/x86_64-linux-gnu/
 ENV LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libjemalloc.so
 
+# stage for arm64
+FROM gcr.io/distroless/cc-debian12:debug AS base-arm64
+
+SHELL ["/busybox/sh", "-c"]
+RUN ln -s /busybox/sh /bin/sh
```

Comment:
anything arch specific about the shell or can this move to final image?

Reply:
We can't do `RUN ln -s /lib/aarch64-linux-gnu/libz.so.1.2.11 /lib/aarch64-linux-gnu/libz.so.1` since this command is executed on shell and it defaults to `/bin/sh`, hence we had to use `SHELL` instruction in base image per arch. I put `ln -s` to `/bin/sh` as well with that